Concepts/ParkingLot/Entities/ParkingLot.cs

```diff
@@ -0,0 +1,158 @@
+        }
+        public AcquiredFloorSlot AcquireFreeSlot(SlotType slotType)
+        {
+            for (int floorId = 1; floorId < _floors.Length; ++floorId)
+            {
+                if (_floors[floorId].SlotTypeFreeSlots[slotType].Count > 0)
+                {
+                    ParkingFloorSlot slot = _floors[floorId].SlotTypeFreeSlots[slotType].ElementAt(0);
+                    _floors[floorId].SlotTypeFreeSlots[slotType].Remove(slot);
+                    _floors[floorId].SlotTypeOccupiedSlots[slotType].Add(slot);
+
+                    return new AcquiredFloorSlot(floorId, slot.SlotId);
+                }
+            }
+            return null;
+        }
+        public void ReleaseSlot(SlotType slotType, int floorId, int slotId)
+        {
+            ParkingFloorSlot slot = new ParkingFloorSlot(slotId);
+            _floors[floorId].SlotTypeOccupiedSlots[slotType].Remove(slot);
+            _floors[floorId].SlotTypeFreeSlots[slotType].Add(slot);
+        }
+        public Dictionary<int, int> GetFreeSlotsCountPerFloor(SlotType slotType)
+        {
+            Dictionary<int, int> floorIdFreeSlotCount = new Dictionary<int, int>();
+            for (int floorId = 1; floorId < _floors.Length; ++floorId)
+            {
+                if (!floorIdFreeSlotCount.ContainsKey(floorId))
+                {
+                    floorIdFreeSlotCount.Add(floorId, 0);
+                }
+                floorIdFreeSlotCount[floorId] += _floors[floorId].SlotTypeFreeSlots[slotType].Count;
+            }
+            return floorIdFreeSlotCount;
+        }
+        public Dictionary<int, List<int>> GetSlotsPerFloor(SlotType slotType, DisplayType displayType)
+        {
+            Dictionary<int, List<int>> floorIdSlotIds = new Dictionary<int, List<int>>();
+            for (int floorId = 1; floorId < _floors.Length; ++floorId)
+            {
+                HashSet<ParkingFloorSlot> slots = displayType == DisplayType.free_slots ?
+                                                                _floors[floorId].SlotTypeFreeSlots[slotType] :
+                                                                _floors[floorId].SlotTypeOccupiedSlots[slotType];
+
+                floorIdSlotIds.Add(floorId, new List<int>());
+                foreach (ParkingFloorSlot parkingFloorSlot in slots)
+                {
+                    floorIdSlotIds[floorId].Add(parkingFloorSlot.SlotId);
+                }
+            }
+            return floorIdSlotIds;
+        }
+
+        private class ParkingFloor
+        {
+            public Dictionary<SlotType, HashSet<ParkingFloorSlot>> SlotTypeFreeSlots;
+            public Dictionary<SlotType, HashSet<ParkingFloorSlot>> SlotTypeOccupiedSlots;
+           
+            public ParkingFloor(int slotsPerFloor)
+            {
+
+                SlotTypeFreeSlots = new Dictionary<SlotType, HashSet<ParkingFloorSlot>>
```

Comment:
We need to allocate the closest slot first, this won't be possible with HashSet, use SortedSet instead.

Reply:
Fixed